### ISSUE TITLE
Generated dependent grammar selects incorrect entry rule

### ISSUE THREAD

**Issue by steve-hickman-epistimis** [TypeFox/xtext2langium#4]
I have several XText grammars with dependencies as follows:
```
grammar com.epistimis.privacy.Privacy with com.epistimis.face.Face
grammar com.epistimis.face.Face with com.epistimis.uddl.Uddl
grammar com.epistimis.uddl.Uddl with org.eclipse.xtext.common.Terminals
```
When I generate from the Privacy grammar, I get the following files:
```
Privacy.langium
Face.langium
Face-types.langium
Uddl.langium
Uddl-types.langium
Terminals.langium
```

The `Face.langium` grammar is generated like this (corrected for issues 2 & 3):
```
grammar Face
import 'Uddl'
//import 'Uddl-types'

entry FaceElement infers FaceElement:
    ArchitectureModel | UopUoPModel | IntegrationIntegrationModel | UopElement 
;
```

However, `Face.xtext` is defined like this:
```
grammar com.epistimis.face.Face with com.epistimis.uddl.Uddl

generate face "http://www.epistimis.com/face/Face"

import "http://www.epistimis.com/uddl/Uddl" as uddl

	
ArchitectureModel:
	'am' name=ID description=STRING '{'
	(dm+=DataModel | um+=UopUoPModel | im+=IntegrationIntegrationModel)+
	'}'
;

FaceElement:
	ArchitectureModel |
	UopUoPModel |
	IntegrationIntegrationModel |
	UopElement
;
```

Note that `ArchitectureModel` is the entry rule for the `Face.xtext` grammar.  It isn't even generated for the `Face.langium` grammar.

**Comment by dhuebner** [comment]
@steve-hickman-epistimis 

> Note that ArchitectureModel is the entry rule for the Face.xtext grammar. It isn't even generated for the Face.langium grammar.

Is `ArchitectureModel` overridden in `com.epistimis.privacy.Privacy`? If so then this ParserRule was removed from  `Face.langium` as otherwise Langium would report in error like: `A rule with the name 'ArchitectureModel' already exists in an imported grammar`

I think it would be a nice fragment option to force-generate overridden rules in super grammars.

**Comment by steve-hickman-epistimis** [comment]
Yes, `ArchitectureModel` is overridden in `com.epistimis.privacy.Privacy`

**Comment by dhuebner** [comment]
@steve-hickman-epistimis 
Langium doesn't support overriding rules from super grammar. You need to refactor your grammar, for example using parser rule parameters to control you super grammar rule. 

I think it would be better to just generate duplicates to see the errors reported by Langium. See also #7

**Comment by steve-hickman-epistimis** [comment]
The challenge there is the usability of the net result.  The base rules are defined in a published standard - so I don't have much room for modifications there - and changes to remove the overriding rules could make the net result less user friendly.

**Comment by dhuebner** [comment]
> The base rules are defined in a published standard - so I don't have much room for modifications there

Currently there is no way to directly override a super grammar rule. 

>  and changes to remove the overriding rules could make the net result less user friendly.

I agree, I will set removing overridden rules to false by default.

**Comment by dhuebner** [comment]
By default overridden rules will not be removed, but you will see an error in the sub grammar and need to resolve it.

**Comment by dhuebner** [comment]
@steve-hickman-epistimis 
I just published version 0.4.0, please try it out. Before starting mwe2 workflow remove or move files in `./langium` first to have a clean state. 
It could be that you need to import transitiv grammar imports like e.g. `terminals.langium` by hand. I'm not sure it is a bug in Langium or expected...
